platform.md

```diff
@@ -869,16 +879,21 @@ For more information on build reproducibility see [https://reproducible-builds.o
 #### `analyzed.toml` (TOML)
 
 ```toml
-[image]
+[previous-image]
   reference = "<image reference>"
-
-[metadata]
-# layer metadata
+[previous-image.metadata]
+  # previous image layer metadata
+[run-image]
+  reference = "<image reference>"
+  mixins = ["libgc", "libpq"]
+[build-image]
+  mixins = ["jq", "libgc", "libpq"]
```

Comment:
```suggestion
[build-image]
  stack-id = "<string>"
  mixins = ["jq", "libgc", "libpq"]
```
I want to add this meanly for symmetry with `stack.toml`. It could be useful eventually if we wanted to ensure that the `detector` and `builder` are in fact running on stack provided to `analyzer`.

Reply:
Should we bubble this up so `stack-id` is shared between build and run? When I tried implementing this in pack, it was a little awkward because we don't usually scope a stack ID to only build or run.

Reply:
@jkutner I could get behind that idea. We should still check that the run-image actually has the declared stack ID though.

Reply:
I added `stack-id` in 6480d2e975ba81d5b61d681f0177539408c834b8

Reply:
Let's go with `[build-image] stack-id`, in this PR and I'll open a follow-up PR to change (then we can debate that separately from this PR).